pkg/v1/tarball/image.go

```diff
@@ -248,6 +259,15 @@ func (i *uncompressedImage) LayerByDiffID(h v1.Hash) (partial.UncompressedLayer,
 			if bd, ok := i.imgDescriptor.LayerSources[h]; ok {
 				// Overwrite the mediaType for foreign layers.
 				mt = bd.MediaType
```

Comment:
You can drop `mt` now and move the `types.DockerLayer` down into the `uncompressedLayerFromTarball` struct literal. Make sure you move the comment with it.

Reply:
Dropped mt. The struct merging has issues as discussed over IM